crates/oxc_linter/src/rules/nextjs/no_typos.rs

```diff
@@ -0,0 +1,211 @@
+use oxc_ast::{
+    ast::{BindingPatternKind, Declaration, ModuleDeclaration},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::phf_set;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-next(no-typos): {0} may be a typo. Did you mean {1}?")]
+#[diagnostic(severity(warning), help("Prevent common typos in Next.js's data fetching functions"))]
+struct NoTyposDiagnostic(String, String, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoTypos;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Prevent common typos in Next.js's data fetching functions
+    ///
+    /// ### Why is this bad?
+    ///
+    ///
+    /// ### Example
+    /// ```javascript
+    /// export default function Page() {
+    ///   return <div></div>;
+    /// }
+    /// export async function getServurSideProps(){};
+    /// ```
+    NoTypos,
+    correctness
+);
+
+const NEXTJS_DATA_FETCHING_FUNCTIONS: phf::Set<&'static str> = phf_set! {
+    "getStaticProps",
+    "getStaticPaths",
+    "getServerSideProps",
+};
+
+// 0 is the exact match
+const THRESHOLD: i32 = 1;
+
+impl Rule for NoTypos {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let Some(path) = ctx.file_path().to_str() else { return };
+        let Some(path_after_pages) = path.split("pages").nth(1) else { return };
```

Comment:
need to figure out how to run test with a file path starting with `/pages/* because this rule only runs when a file is under pages directory

Reply:
https://github.com/oxc-project/oxc/pull/1955

Reply:
got it https://github.com/oxc-project/oxc/pull/1978/commits/59f318946d2b00c736f4c955f8c9d74878224467

Reply:
@camc314 
thanks!
I will fix that and add more test